File: Downloads/expense_manager_app_mian_speaks-master/my_custom_expense_manager/app/src/main/java/com/custom/expensemanager/views/activites/MainActivity.java
package com.custom.expensemanager.views.activites;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.app.AppCompatDelegate;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;
import androidx.recyclerview.widget.LinearLayoutManager;

import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.content.SharedPreferences;

import com.custom.expensemanager.adapters.TransactionsAdapter;
import com.custom.expensemanager.models.Transaction;
import com.custom.expensemanager.utils.Constants;
import com.custom.expensemanager.utils.Helper;
import com.custom.expensemanager.viewmodels.MainViewModel;
import com.custom.expensemanager.views.fragments.AddTransactionFragment;
import com.custom.expensemanager.R;
import com.custom.expensemanager.databinding.ActivityMainBinding;
import com.custom.expensemanager.views.fragments.StatsFragment;
import com.custom.expensemanager.views.fragments.TransactionsFragment;
import com.custom.expensemanager.AccountsFragment;
import com.custom.expensemanager.views.fragments.MoreFragment;
import com.google.android.material.navigation.NavigationBarView;
import com.google.android.material.tabs.TabLayout;

import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Calendar;
import java.util.Date;

import io.realm.Realm;
import io.realm.RealmResults;

public class MainActivity extends AppCompatActivity {

    ActivityMainBinding binding;

    Calendar calendar;
    /*
    0 = Daily
    1 = Monthly
    2 = Calendar
    3 = Summary
    4 = Notes
     */


    public MainViewModel viewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // Restore theme preference
        SharedPreferences sharedPreferences = getSharedPreferences("theme_prefs", MODE_PRIVATE);
        int themeMode = sharedPreferences.getInt("theme_mode", AppCompatDelegate.MODE_NIGHT_FOLLOW_SYSTEM);
        AppCompatDelegate.setDefaultNightMode(themeMode);

        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        viewModel = new ViewModelProvider(this).get(MainViewModel.class);



        setSupportActionBar(binding.toolBar);
        getSupportActionBar().setTitle("Transactions");


        Constants.setCategories();

        calendar = Calendar.getInstance();

        FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
        transaction.replace(R.id.content, new TransactionsFragment());
        transaction.commit();

        binding.bottomNavigationView.setOnItemSelectedListener(new NavigationBarView.OnItemSelectedListener() {
            @Override
            public boolean onNavigationItemSelected(@NonNull MenuItem item) {
                FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
                if(item.getItemId() == R.id.transactions) {
                    getSupportFragmentManager().popBackStack();
                    getSupportActionBar().setTitle("Transactions");
                } else if(item.getItemId() == R.id.stats){
                    transaction.replace(R.id.content, new StatsFragment());
                    transaction.addToBackStack(null);
                    getSupportActionBar().setTitle("Statistics");
                } else if(item.getItemId() == R.id.accounts){
                    transaction.replace(R.id.content, new AccountsFragment());
                    transaction.addToBackStack(null);
                    getSupportActionBar().setTitle("Accounts");
                } else if(item.getItemId() == R.id.more){
                    transaction.replace(R.id.content, new MoreFragment());
                    transaction.addToBackStack(null);
                    getSupportActionBar().setTitle("More");
                }
                transaction.commit();
                return true;
            }
        });


    }

    public void getTransactions() {
        viewModel.getTransactions(calendar);
    }





    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.top_menu,menu);
        return super.onCreateOptionsMenu(menu);
    }
}

File: Desktop/AndroidProject/app/src/main/java/com/example/mentorconnect/ui/main/MainActivity.kt
package com.example.mentorconnect.ui.main

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import com.example.mentorconnect.R
import com.example.mentorconnect.databinding.ActivityMainBinding
import com.example.mentorconnect.ui.main.mentor.MentorListFragment
import com.example.mentorconnect.ui.main.profile.ProfileFragment
import com.example.mentorconnect.ui.main.settings.SettingsFragment

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    private val mentorFragment = MentorListFragment()
    private val profileFragment = ProfileFragment()
    private val settingsFragment = SettingsFragment()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        setSupportActionBar(binding.toolbar)

        binding.bottomNavigation.setOnItemSelectedListener { item ->
            when (item.itemId) {
                R.id.navigation_mentors -> switchFragment(mentorFragment)
                R.id.navigation_profile -> switchFragment(profileFragment)
                R.id.navigation_settings -> switchFragment(settingsFragment)
            }
            true
        }

        if (savedInstanceState == null) {
            binding.bottomNavigation.selectedItemId = R.id.navigation_mentors
        }
    }

    private fun switchFragment(fragment: Fragment) {
        supportFragmentManager.beginTransaction()
            .replace(R.id.container, fragment)
            .commit()
        binding.toolbar.title = when (fragment) {
            is MentorListFragment -> getString(R.string.label_mentors)
            is ProfileFragment -> getString(R.string.label_profile)
            is SettingsFragment -> getString(R.string.label_settings)
            else -> getString(R.string.app_name)
        }
    }
}
